### PR TITLE
Add basic `\d` command to REPL.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Add `\d NAME` command to REPL. It is a shortcut for `DESCRIBE OBJECT`
where the `NAME` is plain text name of the concept that needs to be
described. No backtick name quoting is needed, but it can be used.

For example, the following are equivalent:

    \d O bje`ct
    \d `O bje``ct`

Largely the backticks are optional so that when keywords are used as
names no quoting is necessary. For example, describing a user-created
type `Commit` could be done like this:

    \d Commit
    \d default::Commit
    \d default::`Commit`

Issue: #179

### CODE REVIEW

File: edb/edgeql/parser/grammar/lexer.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ class EdgeQLLexer(lexer.Lexer):
 
         Rule(token='QIDENT',
              next_state=STATE_KEEP,
-             regexp=r'`([^`]|``)*`'),
+             regexp=r'`([^`]|``)+`'),
```
Comment (1st1) [inline edb/edgeql/parser/grammar/lexer.py]: I'm not sure I get these changes. Can you add a couple of syntax tests that wouldn't work with old regexes?

Comment (vpetrovykh) [reply edb/edgeql/parser/grammar/lexer.py]: The change is so that empty string is no longer a valid identifier. It seems like a kind of identifier that is likely to cause errors and confusion. Such as introspected name would be `"modname::"`, which although technically correct, looks more like some kind of error.

I can change it back if we really want to keep empty strings as valid names, I'm just not sure it's a good idea.

I can add some tests for empty string names.

Comment (1st1) [reply edb/edgeql/parser/grammar/lexer.py]: > The change is so that empty string is no longer a valid identifier. 

yeah, add a test for that!

Comment (vpetrovykh) [reply edb/edgeql/parser/grammar/lexer.py]: I've updated the tests and adjusted the regexp back to the original so that the error message about empty identifiers could be more specific.

File: edb/repl/__init__.py [hunk 2]
```diff
@@ -299,6 +301,35 @@ def command_list_dbs(self, args):
         for dbn in result:
             print(f'  {dbn}')
 
+    @_command('d', R'\d NAME', 'describe concept')
+    def command_describe_object(self, args):
+        name = utils.normalize_name(args)
```
Comment (1st1) [inline edb/repl/__init__.py]:
```suggestion
        name = utils.normalize_name(args.strip())
```

File: edb/repl/utils.py [hunk 3]
```diff
@@ -82,3 +84,39 @@ def split_edgeql(script: str, *, script_mode=True):
...
+def _strip_quote(part):
+    if (part != '```'
+            and len(part) > 2
+            and part.startswith('`') and part.endswith('`')):
+        return part[1:-1].replace('``', '`')
+    else:
+        return part
+
+
+def normalize_name(name):
```
Comment (1st1) [inline edb/repl/utils.py]: Type-annotate new functions pls

File: edb/repl/__init__.py (hunk 2 above)
Comment (vpetrovykh) [reply edb/repl/__init__.py]: Done

File: edb/repl/utils.py (hunk 3 above)
Comment (vpetrovykh) [reply edb/repl/utils.py]: Done.
